### PR TITLE
Add docs as required by the Registry

### PR DESCRIPTION
**Pull request by alexsomesan**
### Description

Add documentation compatible with the new Terraform Registry, as generated by https://github.com/hashicorp/terraform-plugin-docs
### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch? (If so, please include the test log in a gist)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+## Getting Started
+
+If this is your first time here, you can get an overview of the provider by reading our [introductory blog post](https://www.hashicorp.com/blog/deploy-any-resource-with-the-new-kubernetes-provider-for-hashicorp-terraform/).
+
+Otherwise, start by downloading a copy of our latest build from the [releases page](https://github.com/hashicorp/terraform-provider-kubernetes-alpha/releases).
```
Comment (aareet) [inline docs/index.md:75]: We don't need this once we're in the registry. In fact we should stop uploading binaries to the github release page for new builds and drive users to the registry and releases.hashicorp.com

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+You'll need to unzip the release download and copy the binary into your [Terraform plugins folder](https://www.terraform.io/docs/plugins/basics.html#installing-plugins) in order for Terraform to discover it.
```
Comment (aareet) [inline docs/index.md:77]: Also not needed

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    }
+  }
+}
+```
+
+### Create a Kubernetes Custom Resource Definition
+
+```hcl
+provider "kubernetes-alpha" {
+  config_path = "~/.kube/config" // path to kubeconfig
```
Comment (aareet) [inline docs/index.md:108]: Maybe we can change this example to one of the other auth methods

File: docs/index.md (hunk 1 above)
Comment (alexsomesan) [reply docs/index.md:75]: Good point, I'll drop it.

**Pull request merged**